plugins/extractors/superset/superset.go

```diff
@@ -0,0 +1,236 @@
+	"github.com/odpf/salt/log"
+	"github.com/pkg/errors"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+//go:embed README.md
+var summary string
+
+// Config hold the set of configuration for the superset extractor
+type Config struct {
+	Username string `mapstructure:"username" validate:"required"`
+	Password string `mapstructure:"password" validate:"required"`
+	Host     string `mapstructure:"host" validate:"required"`
+	Provider string `mapstructure:"provider" validate:"required"`
+}
+
+var sampleConfig = `
+username: meteor_tester
+password: meteor_pass_1234
+host: http://localhost:3000
+provider: db`
+
+// Extractor manages the extraction of data
+// from the superset server
+type Extractor struct {
+	config      Config
+	accessToken string
+	csrfToken   string
+	logger      log.Logger
+	client      *http.Client
+}
+
+// New returns a pointer to an initialized Extractor Object
+func New(logger log.Logger) *Extractor {
+	return &Extractor{
+		logger: logger,
+	}
+}
+
+// Info returns the brief information of the extractor
+func (e *Extractor) Info() plugins.Info {
+	return plugins.Info{
+		Description:  "Dashboard list from Superset server.",
+		SampleConfig: sampleConfig,
+		Summary:      summary,
+		Tags:         []string{"oss", "extractor"},
+	}
+}
+
+// Validate validates the configuration of the extractor
+func (e *Extractor) Validate(configMap map[string]interface{}) (err error) {
+	return utils.BuildConfig(configMap, &Config{})
+}
+
+// Init initializes the extractor
+func (e *Extractor) Init(_ context.Context, configMap map[string]interface{}) (err error) {
+	// build and validate config
+	if err = utils.BuildConfig(configMap, &e.config); err != nil {
+		return plugins.InvalidConfigError{}
+	}
+	e.client = &http.Client{
+		Timeout: 4 * time.Second,
+	}
+	// get access token for further api calls in superset
+	if e.accessToken, err = e.getAccessToken(); err != nil {
+		return errors.Wrapf(err, "failed to get access token")
+	}
+
+	// get csrf token for secure api calls in superset
+	if e.csrfToken, err = e.getCsrfToken(); err != nil {
+		return errors.Wrapf(err, "failed to get csrf token")
+	}
+	return
+}
+
+// Extract collects metadata of each dashboard through emitter
+func (e *Extractor) Extract(_ context.Context, emit plugins.Emit) (err error) {
+	dashboards, err := e.getDashboardsList()
+	if err != nil {
+		return errors.Wrapf(err, "failed to get dashboard list")
+	}
+	for _, dashboard := range dashboards {
+		data, err := e.buildDashboard(dashboard.ID)
+		if err != nil {
+			return errors.Wrapf(err, "failed to build dashbaord")
+		}
+		emit(models.NewRecord(data))
+	}
+	return
+}
+
+// buildDashboard builds a dashboard from superset server
+func (e *Extractor) buildDashboard(id int) (data *assets.Dashboard, err error) {
+	var dashboard Dashboard
+	chart, err := e.getChartsList(id)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get chart list")
+	}
+
+	var tempCharts []*assets.Chart
```

Comment:
can we move this mapping to `getChartsList` as well.
so this `getChartsList` will `assets.Chart` instead of superset's chart.